### DISCUSSION TITLE
Why  doesn't  Input field changed ? Seeking Help with React Router Tutorial

### DISCUSSION THREAD

**Discussion by 2xx** [remix-run/react-router#11198] [Q&A]
Hey everyone,

I'm currently going through a tutorial on React Router, and I've come across an issue that's puzzling me. The tutorial link is here.

The specific problem (Problem 1) I'm facing is a bit confusing. I initially thought I had resolved Problem 2, where changing the URL updates the input box's defaultValue. However, when I click the browser's back button, the value of the input search box doesn't revert. I've checked the variable 'q' in the component, and it does change, but it just doesn't update in the form.

I suspect there might be some technical details that I'm not familiar with. I was hoping someone could provide some insight or guidance on this issue.

Thanks in advance for any help!

**Comment by kiliman** [comment]
I don't see a link. Can you post the link to the tutorial?

**Reply by 2xx** [reply]
1. I'm sorry, I'm not good at using the current form of communication. After a few days, I forgot about the questions here. I apologize again.

2. Regarding my question, I think it's because of defaultValue. I wasn't familiar with the difference between the value and defaultValue of an input box before, but now I understand it better.

3. I clumsily forgot to provide specific links in my previous question, and I'm adding them now:
    https://reactrouter.com/en/main/start/tutorial#synchronizing-urls-to-form-state
